packages/dbgpt-ext/src/dbgpt_ext/rag/knowledge/git_repo.py

```
@@ -0,0 +1,422 @@
+        self._clone_dir = tempfile.mkdtemp(prefix="git_repo_knowledge_")
+        cmd = ["git", "clone", "--branch", self._branch]
+        if shallow:
+            cmd.extend(["--depth", "1"])
+        cmd.extend([self._repo_url, self._clone_dir])
+
+        try:
+            logger.info(
+                f"Cloning repo {self._repo_url} branch={self._branch} "
+                f"shallow={shallow} to {self._clone_dir}"
+            )
+            subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)